test/smoke/FirefoxPrivateVPNUITest/FirefoxPrivateVPNUITest/Common/UserCommonOperation.cs

```diff
@@ -178,28 +191,34 @@ public static void DisconnectVPN(FirefoxPrivateVPNSession vpnClient, DesktopSess
         /// <param name="client">RestClient object.</param>
         /// <param name="request">RestRequest object.</param>
         /// <returns>The API response.</returns>
-        private static IRestResponse RetryExecute(IRestClient client, IRestRequest request)
+        private static IRestResponse RetryExecute(IRestClient client, IRestRequest request, Func<IRestResponse, bool> condition)
         {
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+            int retries = 0;
+
             // Define retry policy
-            var policy = Policy.HandleResult<IRestResponse>((response) =>
-            {
-                return response.StatusCode != HttpStatusCode.OK;
-            }).WaitAndRetry(3, (count) => TimeSpan.FromSeconds(10));
+            var policy = Policy.HandleResult(condition).WaitAndRetry(5, (count) => TimeSpan.FromSeconds(10));
 
             // Execute the request with policy
             var val = policy.ExecuteAndCapture(() =>
             {
-                return client.Execute(request);
+                Console.WriteLine($"{client.BaseUrl} - sent times: {++retries}");
+                var response = client.Execute(request);
+                Console.WriteLine($"Response: {response.StatusCode} {response.Content}");
+                return response;
             });
-            IRestResponse rr = val.Result;
 
+            IRestResponse rr = val.Result;
             if (rr == null)
             {
                 rr = new RestResponse();
                 rr.Request = request;
                 rr.ErrorException = val.FinalException;
             }
 
+            stopwatch.Stop();
+            Console.WriteLine($"The total time to get the expected response: {stopwatch.ElapsedMilliseconds} milliseconds.");
```

Comment:
This might pollute your log. Suggest you log only when the operation in question had to retry or if the operation was unusually long.

Reply:
This will be temporary log and all of console log might be removed if everything works fine for a while. QA wants to know how long it takes to get correct response from Mullvad API after connection. Once any errors regarding Mullvad API occur, we need to provide some data to Mullvad and tell them how long it takes for our vpn client to connect. This log also help QA to debug the automation test if the test fails on the API call to Mullvad.